File: qe-selenium-python/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
driver.maximize_window()

try:
    driver.get("https://training-support.net/webelements/dynamic-content")
    print("Page Title:", driver.title)

    wait = WebDriverWait(driver, 10)

    click_me = wait.until(
        EC.element_to_be_clickable((By.XPATH, "//button[text()='Click me']"))
    )
    click_me.click()

    text_element = wait.until(
        EC.visibility_of_element_located((By.ID, "dynamicText"))
    )
    print("Displayed Text:", text_element.text)

finally:
    driver.quit()

File: qe-java-selenium/Activity12.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;

public class Activity12 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(20));

        try {
            driver.get("https://training-support.net/webelements/dynamic-content");
            
            // 1. Find and click the button
            WebElement clickMeButton = wait.until(ExpectedConditions.elementToBeClickable(By.id("genButton")));
            clickMeButton.click();
            System.out.println("Button clicked. Waiting for 'release' word...");

            // 2. Custom Wait: Wait until the button text (in lowercase) contains "release"
            // This ignores whether the site says "RELEASE", "Release", or "release"
            wait.until(d -> clickMeButton.getText().toLowerCase().contains("release"));

            // 3. Get the text and print it
            String finalText = clickMeButton.getText();
            System.out.println("Text found: " + finalText);

        } catch (Exception e) {
            System.out.println("Error: " + e.getMessage());
        } finally {
            // 4. Close the browser
            driver.quit();
        }
    }
}